lib/cli-engine.js

```diff
@@ -162,14 +165,33 @@ function processFile(filename, configHelper) {
         config,
         text,
         messages,
-        stats;
+        stats,
+        fileExtension = path.extname(filename).substring(1),
+        preprocessor;
 
     if (fs.existsSync(filePath)) {
         debug("Linting " + filePath);
         config = configHelper.getConfig(filePath);
         loadPlugins(config.plugins);
         text = fs.readFileSync(path.resolve(filename), "utf8");
-        messages = eslint.verify(text, config, filename);
+
+        for (var plugin in loadedPlugins) {
+            if (loadedPlugins[plugin].preprocessor && Object.keys(loadedPlugins[plugin].preprocessor).indexOf(fileExtension) >= 0) {
```

Comment:
Woops. Good catch. That's the problem with the unittest I wrote, it bypasses traverse function, so it not going to reject .txt file, even if preprocessor is not registered. (which is the case with feeding a single line to eslint through command line, btw).